Bottom: e257627a370af21d61a96c599a7b2aedd817e45d
Top:    ec5fd595dbc7ad2aa6e105e65257fb2c886f7c50
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-03-26 16:13:14 +0000

Remove hack in draw.c which broke PS (and probably other) output

The set_layer call hacked into DrawLayerGroup to reset compositing tricked
the PS exporter to re-open and overwrite output in the main layer output
file during the fab drawing output.


---

diff --git a/src/draw.c b/src/draw.c
index 40b11ec..bed42e1 100644
--- a/src/draw.c
+++ b/src/draw.c
@@ -882,8 +882,7 @@ DrawLayerGroup (int group, const BoxType * screen)
   Cardinal *layers = PCB->LayerGroups.Entries[group];
 
   clip_box = screen;
-  for (i = n_entries - 1; i >= 0;
-      i--, gui->set_layer (0, group, 0)) /* HACK: Subcomposite each layer in a layer group separately */
+  for (i = n_entries - 1; i >= 0; i--)
     {
       layernum = layers[i];
       Layer = PCB->Data->Layer + layers[i];
@@ -900,10 +899,6 @@ DrawLayerGroup (int group, const BoxType * screen)
 	      r_search (Layer->polygon_tree, screen, NULL, poly_callback,
 			&info);
 	      info.arg = false;
-
-	      /* HACK: Subcomposite polygons separately from other layer primitives */
-	      /* Reset the compositing */
-	      gui->set_layer (0, group, 0);
 	    }
 
 	  if (TEST_FLAG (CHECKPLANESFLAG, PCB))
